jgrapht-io/src/main/java/org/jgrapht/nio/tsplib/TSPLIBImporter.java

```diff
@@ -0,0 +1,735 @@
+ * cities of different countries.
+ * </p>
+ * 
+ * @author Hannes Wellmann
+ * @param <V> the type of vertices created by this importer
+ *
+ */
+public class TSPLIBImporter<V>
+    implements
+    GraphImporter<V, DefaultWeightedEdge>
+{
+    /**
+     * Container for data of an imported <em>TSPLIB95</em> file.
+     * 
+     * @author Hannes Wellmann
+     * @param <V> the type of vertices in the imported graph
+     */
+    public static class TSPLIBFileData<V>
+    {
+
+        private String name;
+        private String type;
+        private String comment;
+        private Integer dimension;
+        private String edgeWeightType;
+        private String nodeCoordType;
+        private boolean hasDistinctLocations;
+        private Boolean hasDistinctEdges;
+
+        private Graph<V, DefaultWeightedEdge> graph;
+        private List<Integer> tour;
+
+        // getters for read public data
+
+        /**
+         * Returns the value of the <em>NAME</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>NAME</em> keyword
+         */
+        public String getName()
+        {
+            return name;
+        }
+
+        /**
+         * Returns the value of the <em>TYPE</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>TYPE</em> keyword
+         */
+        public String getType()
+        {
+            return type;
+        }
+
+        /**
+         * Returns the joined value of the <em>COMMENT</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>COMMENT</em> keyword
+         */
+        public String getComment()
+        {
+            return comment;
+        }
+
+        /**
+         * Returns the value of the <em>DIMENSION</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>DIMENSION</em> keyword
+         */
+        public Integer getDimension()
+        {
+            return dimension;
+        }
+
+        /**
+         * Returns the value of the <em>EDGE_WEIGHT_TYPE</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>EDGE_WEIGHT_TYPE</em> keyword
+         */
+        public String getEdgeWeightType()
+        {
+            return edgeWeightType;
+        }
+
+        /**
+         * Returns the value of the <em>NODE_COORD_TYPE</em> keyword in the imported file.
+         * 
+         * @return the value of the <em>NODE_COORD_TYPE</em> keyword
+         */
+        public String getNodeCoordType()
+        {
+            return nodeCoordType;
+        }
+
+        /**
+         * Returns the complete {@link Graph} of location connections built from the imported file
+         * or null if no graph was imported.
+         * 
+         * @return the complete {@code Graph} of location connections
+         */
+        public Graph<V, DefaultWeightedEdge> getGraph()
```

Comment:
Remove this method.  We have a clear interface on all importers, which is to the `importGraph` method. No need to return the graph using two different ways.